### ISSUE TITLE
noise_loss

### ISSUE THREAD

**Issue by fregu856** [ruqizhang/csgmcmc#1]
Hi,

Thanks for an interesting paper and open-source code!

I'm struggling somewhat to match your SGLD/SGHMC implementation with the update equations in the paper. In https://github.com/ruqizhang/csgmcmc/blob/51e511478d607b2523fc803d82a26edd39b14b6d/experiments/cifar_csgmcmc.py (line 79), you have ```noise_std = 2*lr*alpha**0.5```. We will take the gradient of this noise term and then multiply the result with lr, right? Will that actually match the update equation for SGLD then?

Here's how I would implement SGLD (with p(theta) = N(0, alpha*I)), N=datasize):
```
loss_fn = nn.CrossEntropyLoss()

loss_likelihood = loss_fn(logits, y)

loss_prior = 0.0
for param in network.parameters():
    loss_prior += (1.0/2.0)*(1.0/N)*(1.0/alpha)*torch.sum(torch.pow(param, 2))

loss_noise = 0.0
for param in network.parameters():
    loss_noise += (1.0/math.sqrt(N))*math.sqrt(2.0/lr)*torch.sum(param*Variable(torch.normal(torch.zeros(param.size()), std=1.0).cuda()))

loss = loss_likelihood + loss_prior + loss_noise
```

Hope you can help resolve my confusion.

Regards

//
Fredrik

**Comment by ruqizhang** [comment]
Hi Fredrik, thank you for your interest. We have fixed this bug.

**Comment by HuangZiYang2000** [comment]
Hi! I am still confused with the noise_std formula. Why you change the code from `noise_std = 2*lr*alpha**0.5` to `noise_std = (2/lr*alpha)**0.5`? I thought the noise_std should decrease as the lr decreasing. 
Sorry for bothering you after such long time!
